src/instrumentsscene/qml/MuseScore/InstrumentsScene/InstrumentsDialog.qml

```diff
@@ -25,10 +25,12 @@ import QtQuick.Layouts 1.15
 import Muse.Ui 1.0
 import Muse.UiComponents 1.0
 import MuseScore.InstrumentsScene 1.0
+import MuseScore.Project 1.0
 
 StyledDialogView {
     id: root
 
+    required property InstrumentsOnScoreListModel instrumentsOnScoreModel
```

Comment:
This will not work: `InstrumentsDialog` is not instantiated directly from QML, but as a dialog. Therefore, it won't be possible to pass anything to this required property, and the dialog will fail to open. 

Instead, you will need to instantiate an `InstrumentsOnScoreListModel` here inside `InstrumentsDialog`, just like you do in `ChooseInstrumentsAndTemplatesPage`.